proofread: stats_watcher.md (#1583)
juicedata/juicefs: docs/en/benchmark/stats_watcher.md

---

> # JuiceFS Performance Statistics Watcher

Reviewer: ```suggestion
# JuiceFS Performance Statistics Monitor
```
用monitor会不会好一些？感觉watcher有些怪怪的？

> JuiceFS exposes a lot of [Promethues metrics](../administration/monitoring.md) for monitoring system internal performance. However, when diagnosing performace issues in practice, users may need a more real-time monitorig tool to know what is actually going on within a certain time range. Thus, we provide a command `stats` to print metrics every second, just like what the Linux command `dstat` does. The output is like:

Reviewer: ```suggestion
JuiceFS uses [Promethues API for monitoring system internal performance](../administration/monitoring.md). However, when diagnosing performance issues in practice, users may need a more real-time monitoring tool to know what is actually going on within a certain time range. Thus, we provide a command `stats` to print metrics every second, just like what the Linux command `dstat` does. The output is like:
```

Author: 啊，这样感觉意思变了。Prometheus 是一个监控系统，会实时自动访问所有被监控者（比如 JuiceFS）的 API 来获取监控数据，所以其实监控者是 Prometheus，而 JuiceFS 只是暴露了 API 给 Prometheus。另外两个拼写错误我是真没注意，我改一下哈哈

![stats_watcher](../images/juicefs_stats_watcher.png)

> By default, this command will print the following metrics of the JuiceFS process corresponding to the given mount point.

Reviewer: ```suggestion
By default, this command will print the following metrics of the JuiceFS process in the given mount point.
```

Author: emm，感觉这样意思就变了。我理解的是，每个 mount point 都有一个（或多个） JuiceFS process 在维持其功能，而 this  command 打印的是这些个 process 输出的内容，而不是「输出到 given mount point 里」

#### fuse

- ops/lat: operations processed by FUSE per second, and their average latency (in milliseconds)
- read/write: read/write bandwidth usage of FUSE

#### meta

> - ops/lat: metadata operations processed per second, and their average latency (in milliseconds). Please note that, operations returned directly from cache are not counted in, in order to show a more accurate latency of clients actually interacting with metadata engine.

Reviewer: ```suggestion
- ops/lat: metadata operations processed per second, and their average latency (in milliseconds). Please note that operations returned directly by cache are not counted in order to show a more accurate latency of clients interacting with metadata engine.
```
- 如果是缓存直接处理的元数据请求，这里用by是不是更好一些？
- a more accurate latency 好像已经包含了 actual的意思

Author: > 如果是缓存直接处理的元数据请求，这里用by是不是更好一些？

我理解的，数据应该是储存在 cache 里，但返回数据应该是 JuiceFS process 的功能，所以 from cache 更合适一些吧

> a more accurate latency 好像已经包含了 actual的意思

我也不太确定这里该不该加上 actually。我加了主要是想强调的是，这些 interacting 是 client 直接和 server 交互的，而不算那些从 cache 里来的。其实感觉去掉 actually 也可以，咱俩可以 slack 讨论下 😁
